### ISSUE TITLE
Remove Bindings

### ISSUE THREAD

**Issue by 4uva4ok1905** [jonataslaw/getx#1746]
Hi, update from 4.1.4 -> 4.3.5, and I have problem with remove other Class (but Controllers is work fine) after Get.offNamed

4.1.4 - Working normal
FIRST VISIT FAVORITE PAGE
```
[GETX] Instance "GetMaterialController" has been created
[GETX] Instance "GetMaterialController" has been initialized
[GETX] GOING TO ROUTE /favorite
[GETX] Instance "FavoriteProvider" has been created
[GETX] Instance "FavoriteProvider" has been initialized
[GETX] Instance "FavoriteRepository" has been created
[GETX] Instance "FavoriteController" has been created
[GETX] Instance "FavoriteController" has been initialized
```

GO TO PROFILE PAGE - Get.offNamed(Routes.PROFILE);
```
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] Instance "ProfileController" has been created
[GETX] Instance "ProfileController" has been initialized
[GETX] "FavoriteProvider" onDelete() called
[GETX] "FavoriteProvider" deleted from memory
[GETX] "FavoriteRepository" deleted from memory
[GETX] "FavoriteController" onDelete() called
[GETX] "FavoriteController" deleted from memory

```

4.3.5
FIRST VISIT FAVORITE PAGE
```
[GETX] Instance "GetMaterialController" has been created
[GETX] Instance "GetMaterialController" has been initialized
[GETX] GOING TO ROUTE /favorite
[GETX] Instance "FavoriteProvider" has been created
[GETX] Instance "FavoriteProvider" has been initialized
[GETX] Instance "FavoriteRepository" has been created
[GETX] Instance "FavoriteController" has been created
[GETX] Instance "FavoriteController" has been initialized
```

GO TO PROFILE PAGE - Get.offNamed(Routes.PROFILE);
```
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] Instance "ProfileController" has been created
[GETX] Instance "ProfileController" has been initialized
[GETX] "FavoriteController" onDelete() called
[GETX] "FavoriteController" deleted from memory
```

**Why FavoriteProvider && FavoriteRepository not delete from memory?**

Minimal working  version
```
import 'package:flutter/material.dart';
import 'package:get/get.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(
      debugShowCheckedModeBanner: false,
      initialRoute: AppPages.INITIAL,
      getPages: AppPages.routes,
    );
  }
}

class Routes {
  static const FAVORITE = '/favorite';
  static const PROFILE = '/profile';
}

class AppPages {
  static const INITIAL = Routes.FAVORITE;

  static final routes = [
    GetPage(
      name: Routes.FAVORITE,
      page: () => FavoriteScreen(),
      binding: FavoriteBinding(),
    ),
    GetPage(
      name: Routes.PROFILE,
      page: () => ProfileScreen(),
      binding: ProfileBinding(),
    ),
  ];
}

class FavoriteProvider extends GetConnect {}

class FavoriteRepository {
  FavoriteRepository({required this.favoriteProvider});

  final FavoriteProvider favoriteProvider;
}

class FavoriteBinding extends Bindings {
  @override
  void dependencies() {
    Get.put<FavoriteProvider>(FavoriteProvider());
    Get.put<FavoriteRepository>(
      FavoriteRepository(favoriteProvider: Get.find()),
    );

    Get.put<FavoriteController>(FavoriteController());
  }
}

class FavoriteController extends GetxController {}

class FavoriteScreen extends GetView<FavoriteController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('FAVORITE'),
          GestureDetector(
            onTap: () {
              Get.offNamed(Routes.PROFILE);
            },
            child: Text('Go to PROFILE'),
          )
        ],
      ),
    );
  }
}

class ProfileBinding extends Bindings {
  @override
  void dependencies() {
    Get.put<ProfileController>(ProfileController());
  }
}

class ProfileController extends GetxController {}

class ProfileScreen extends GetView<ProfileController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('PROFILE'),
          GestureDetector(
            onTap: () {
              Get.offNamed(Routes.FAVORITE);
            },
            child: Text('Go to FAVORITE'),
          )
        ],
      ),
    );
  }
}
```

**Comment by jaysavsani07** [comment]
facing same issue with 4.3.4 & 4.3.5. any idea how to fix this? @4uva4ok1905

**Comment by csnguyen-gmail** [comment]
same issue here

**Comment by jonataslaw** [comment]
Fixed in get 4.3.6
